relayer/chains/cosmos/tx.go

```diff
@@ -1676,21 +1673,16 @@ func (cc *CosmosProvider) PrepareFactory(txf tx.Factory, signingKey string) (tx.
 }
 
 // AdjustEstimatedGas adjusts the estimated gas usage by multiplying it by the gas adjustment factor
-// and bounding the result by the maximum gas amount option. If the gas usage is zero, the adjusted gas
-// is also zero. If the gas adjustment factor produces an infinite result, an error is returned.
-// max-gas-amount is enforced.
+// and return estimated gas is higher than max gas error. If the gas usage is zero, the adjusted gas
+// is also zero.
 func (cc *CosmosProvider) AdjustEstimatedGas(gasUsed uint64) (uint64, error) {
 	if gasUsed == 0 {
 		return gasUsed, nil
 	}
-	gas := cc.PCfg.GasAdjustment * float64(gasUsed)
-	if math.IsInf(gas, 1) {
-		return 0, fmt.Errorf("infinite gas used")
-	}
-	// Bound the gas estimate by the max_gas option
-	if cc.PCfg.MaxGasAmount > 0 {
-		gas = math.Min(gas, float64(cc.PCfg.MaxGasAmount))
+	if cc.PCfg.MaxGasAmount > 0 && gasUsed > cc.PCfg.MaxGasAmount {
+		return 0, fmt.Errorf("estimated gas %d is higher than max gas %d", gasUsed, cc.PCfg.MaxGasAmount)
 	}
+	gas := cc.PCfg.GasAdjustment * float64(gasUsed)
```

Comment:
this change looks find but do you think we should leave the check from `math.IsInf` in place? It is possible that a user accidentally, or without realizing the implications, sets the value of `GasAdjustment` to some insanely large integer value and then we could end up with `MaxFloat64` from `cc.PCfg.GasAdjustment * float64(gasUsed)`